src/basicfuns.jl

```diff
@@ -286,6 +291,9 @@ function logmxp1(x::Float64)
     end
 end
 
+# Naive fallback
+logmxp1(x::Real) = (log(x) + one(x)) - x
```

Comment:
Should we use
```suggestion
logmxp1(x::Real) = log1pmx(x - one(x))
```
to take advantage of a (possibly specialized) implementation of `log1pmx`? That's also used in the branch for `x > 0.6` in the implementation for `Float64`.

Reply:
Right now, this fails for very small x:
```julia
julia> log1pmx(1e-100 - 1)
-Inf

julia> logmxp1(1e-100)
-229.25850929940458

julia> log(1e-100) + 1 - 1e-100
-229.25850929940458
```

Reply:
And for `Float32`:
```julia

julia> logmxp1(1f-40)
-91.10341f0

julia> log1pmx(1f-40 - 1)
-Inf32
```

Reply:
I guess that's why it's not used for small values for `Float64` :smile: 

Maybe we could use a very rough heuristic such as
```suggestion
function logmxp1(x::Real)
    one_x = one(x)
    if 2 * x < one_x
        # for small values of `x` the other branch returns non-finite values
        (log(x) + one_x) - x
    else
        log1pmx(x - one_x)
    end
end
```